CASMPET-5593 Update spire postgres without backup restore doc

kburns-hpe opened the pull request:
This fixes some issues where spacing would cause the command to try to run rm / if you copy/pasted it. It also replaces the sysmgmt hand-edits with a for loop that will reduce errors and prevent the sysmgmt.yaml that's saved in the loftsman namespace from being overwritten. Commands were tested on Loki.

mharding-hpe commented on operations/spire/Restore_Spire_Postgres_without_a_Backup.md:
```diff
@@ -50,40 +47,38 @@ The CSM release tarball is required as it contains the Spire helm chart.
 3. Get the current cached sysmgmt manifest.
```
```suggestion
3. Get the current cached `sysmgmt` manifest.
```

mharding-hpe commented on operations/spire/Restore_Spire_Postgres_without_a_Backup.md:
````diff
@@ -50,40 +47,38 @@ The CSM release tarball is required as it contains the Spire helm chart.
    ```
 
-4. Edit the `sysmgmt.yaml` spec.charts section to only include the `spire` chart and all its current data. (The resources specified above will be updated in the next step and the version may differ, because this is an example).
+4. Run the following command to remove non-spire charts from the spire.yaml file. This will also change the metadata.name so that it doesn't overwrite the sysmgmt.yaml file that is stored in the loftsman namespace.
````
```suggestion
4. Run the following command to remove non-Spire charts from the `spire.yaml` file.

   This will also change the `metadata.name` so that it does not overwrite the `sysmgmt.yaml` file that is stored in the `loftsman` namespace.
```

mharding-hpe commented on operations/spire/Restore_Spire_Postgres_without_a_Backup.md:
````diff
@@ -50,40 +47,38 @@ The CSM release tarball is required as it contains the Spire helm chart.
+4. Run the following command to remove non-spire charts from the spire.yaml file. This will also change the metadata.name so that it doesn't overwrite the sysmgmt.yaml file that is stored in the loftsman namespace.
 
    ```bash
+   for i in $(yq r spire.yaml 'spec.charts[*].name' | grep -Ev '^spire$'); do yq d -i spire.yaml  'spec.charts(name=='"$i"')'; done; yq w -i spire.yaml metadata.name spire
````
```suggestion
   ncn# for i in $(yq r spire.yaml 'spec.charts[*].name' | grep -Ev '^spire$'); do
                yq d -i spire.yaml  'spec.charts(name=='"$i"')'
            done ; yq w -i spire.yaml metadata.name spire
```

mharding-hpe commented on operations/spire/Restore_Spire_Postgres_without_a_Backup.md:
````diff
@@ -50,40 +47,38 @@ The CSM release tarball is required as it contains the Spire helm chart.
+   for i in $(yq r spire.yaml 'spec.charts[*].name' | grep -Ev '^spire$'); do yq d -i spire.yaml  'spec.charts(name=='"$i"')'; done; yq w -i spire.yaml metadata.name spire
+   ```
+
+   Example spire.yaml after the command is run:
````
```suggestion
   Example `spire.yaml` after the command is run:
```

mharding-hpe commented on operations/spire/Restore_Spire_Postgres_without_a_Backup.md:
````diff
@@ -50,40 +47,38 @@ The CSM release tarball is required as it contains the Spire helm chart.
+
+   Example spire.yaml after the command is run:
+
+   ```text
````
```suggestion
   ```yaml
```

mharding-hpe commented on operations/spire/Restore_Spire_Postgres_without_a_Backup.md:
````diff
@@ -50,40 +47,38 @@ The CSM release tarball is required as it contains the Spire helm chart.
+   ncn# manifestgen -c customizations.yaml -i spire.yaml -o manifest.yaml
    ```
 
 6. Update the helm chart path in the manifest.yaml file.
````
```suggestion
6. Update the helm chart path in the `manifest.yaml` file.
```

mharding-hpe commented on operations/spire/Restore_Spire_Postgres_without_a_Backup.md:
````diff
@@ -93,7 +88,7 @@ The CSM release tarball is required as it contains the Spire helm chart.
    ```
 
 7. Validate that the manifest.yaml file only contains chart information for Spire,
````
```suggestion
7. Validate that the `manifest.yaml` file only contains chart information for Spire,
```

mharding-hpe commented on operations/spire/Restore_Spire_Postgres_without_a_Backup.md:
```diff
@@ -93,7 +88,7 @@ The CSM release tarball is required as it contains the Spire helm chart.
 
 7. Validate that the manifest.yaml file only contains chart information for Spire,
-and that the sources charts location points to the directory the helm chart was extracted from to prepend to /helm.
+   and that the sources charts location points to the directory the helm chart was extracted from to prepend to /helm.
```
I don't know if this is what the sentence meant, but as it was written, it did not make much sense. 
```suggestion
   and that the source charts location points to the directory that the helm chart was extracted from.
```

mharding-hpe reviewed (changes requested):
These changes apply to the other versions of this PR as well.

kburns-hpe commented:
@mharding-hpe I changed some of the instructions and fixed the issues you pointed out. Can you take another look at these? They're identical between the three branches, besides the spire.yaml example's spire chart version.

mharding-hpe commented on operations/spire/Restore_Spire_Postgres_without_a_Backup.md:
````diff
@@ -13,104 +12,97 @@ event that spire-postgres databases cannot be restored from a backup.
    ```
 
-3. Disable spire-agent on all of the Kubernetes NCNs (all worker nodes and master nodes) and delete the join data.
+3. Disable spire-agent on all of the Kubernetes NCNs (all worker nodes and
````
```suggestion
3. Disable `spire-agent` on all of the Kubernetes NCNs (all worker nodes and
```

mharding-hpe commented on operations/spire/Restore_Spire_Postgres_without_a_Backup.md:
```diff
@@ -13,104 +12,97 @@ event that spire-postgres databases cannot be restored from a backup.
-and that the sources charts location points to the directory the helm chart was extracted from to prepend to /helm.
+1. Validate that the `manifest.yaml` file only contains chart information for
+   Spire, and that the sources chart location points to
+   `https://packages.local/repository/charts`
```
```suggestion
   `https://packages.local/repository/charts`.
```

mharding-hpe commented on operations/spire/Restore_Spire_Postgres_without_a_Backup.md:
````diff
@@ -13,104 +12,97 @@ event that spire-postgres databases cannot be restored from a backup.
    ```bash
-   ncn# kubectl get secrets -n loftsman site-init -o jsonpath='{.data.customizations\.yaml}' | base64 -d > customizations.yaml
+   ncn# kubectl get cm -n loftsman loftsman-sysmgmt -o jsonpath='{.data.manifest\.yaml}' > spire.yaml
+
````
```suggestion
```
